### PR TITLE
[libmicrohttpd] Correctly compute size of `timeval.tv_sec` on Windows.

### PR DESCRIPTION
**Pull request by mgautierfr**
On Windows, `timevale.tv_sec` is not a `time_t` but a long [1].
Libmicrohttpd check this value at runtime with assert. [2]

`prefix` is a copy of prefix used in upstream `configure.ac` plus the

   #ifdef _WIN32
   #include <zinsock.h>
   #endif

added.

[1] https://learn.microsoft.com/en-us/windows/win32/api/winsock/ns-winsock-timeval
[2] src/microhttpd/daemon.c, line 8210

### CODE REVIEW

**Comment by mgautierfr** [comment]
I have added the release "0.9.77-3".

Is it possible to also have a "0.9.76-4" ? Simply add the version in `releases.json` ?

File: subprojects/packagefiles/libmicrohttpd/meson.build [hunk 1]
```diff
@@ -118,7 +118,25 @@ foreach s : ['size_t', 'unsigned int', 'unsigned long long']
   cdata.set('SIZEOF_@0@'.format(s.underscorify().to_upper()), cc.sizeof(s))
 endforeach
 
-cdata.set('SIZEOF_STRUCT_TIMEVAL_TV_SEC', cc.sizeof('time_t', prefix: '#include <time.h>'))
+cdata.set(
```
Comment (neheb) [inline subprojects/packagefiles/libmicrohttpd/meson.build:121]: this is garbage and will not work.

Comment (mgautierfr) [reply subprojects/packagefiles/libmicrohttpd/meson.build:121]: Well, it works when compiling natively on Windows, Linux and MacOs.
And cross compiling from Linux 64bits to Linux 32 bits, arm (32 and 64 bits), Anroid (various archs).

And it is what is done by upstream libmicrohttpd autotools: https://github.com/Karlson2k/libmicrohttpd/blob/master/configure.ac#L2448-L2474

So can you explain a bit more "garbage" and "will not work" ?

Comment (kelson42) [reply subprojects/packagefiles/libmicrohttpd/meson.build:121]: > this is garbage and will not work.

Extremly rude and inefficient feedback!

Comment (neheb) [reply subprojects/packagefiles/libmicrohttpd/meson.build:121]: You ported an autotools check thinking meson works the same as autotools. Why do you think those macros for those headers are available?

Just switch the check from time_t to long.

Comment (mgautierfr) [reply subprojects/packagefiles/libmicrohttpd/meson.build:121]: > Why do you think those macros for those headers are available?

Which macro are you speaking about ?

> Just switch the check from time_t to long.

`long` is not the right type outside of Windows.
My first test was indeed to use `long` when on Windows but it seems proper to me to test the size of `timeval.tv_sec` whatever is its real type to be platform independent.

Comment (benoit-pierre) [reply subprojects/packagefiles/libmicrohttpd/meson.build:121]: > > Why do you think those macros for those headers are available?
> 
> Which macro are you speaking about ?

All those `HAVE_xxx`: they're not defined, so those `#ifdef HAVE_[…] #endif` blocks are dead code.

Comment (mgautierfr) [reply subprojects/packagefiles/libmicrohttpd/meson.build]: Ok. New commit (to fixup) simply use `long` on Windows.

**Pull request merged**